stats.go

```diff
@@ -393,6 +466,18 @@ func (s *statStore) Flush() {
 		return true
 	})
 
+	s.timers.Range(func(key, v interface{}) bool {
+		if timer, ok := v.(*reservoirTimer); ok {
+			sampleRate := timer.SampleRate()
+			for _, value := range timer.CollectedValue() {
+				s.sink.FlushAggregatedTimer(key.(string), value, sampleRate)
+			}
+			s.timers.Delete(key) // delete it from the map so it's not flushed again
```

Comment:
Deletion is expensive and since we assume the names of stats are stable will just lead to it being re-added to the map at a later time. For counters we only flush non-zero values and I think we could do something similar with the `reservoirTimer` type (say not flush if it has zero values and trim the slice to zero elements on flush).